web/courses.go

```diff
@@ -97,25 +97,22 @@ func (s *QuickFeedService) rejectEnrollment(ctx context.Context, sc scm.SCM, enr
 }
 
 // enrollStudent enrolls the given user as a student into the given course.
-func (s *QuickFeedService) enrollStudent(ctx context.Context, sc scm.SCM, enrolled *qf.Enrollment) error {
+func (s *QuickFeedService) enrollStudent(ctx context.Context, sc scm.SCM, query *qf.Enrollment) error {
 	// course and user are both preloaded, no need to query the database
-	course, user := enrolled.GetCourse(), enrolled.GetUser()
+	course, user := query.GetCourse(), query.GetUser()
 
 	// check whether user repo already exists,
 	// which could happen if accepting a previously rejected student
 	repo, err := s.getRepo(course, user.GetID(), qf.Repository_USER)
 	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 		return fmt.Errorf("failed to get %s repository for %q: %w", course.Code, user.Login, err)
 	}
-
+	// Use enrollment with full updated info to ensure that gorm Select.Updates works correctly.
```

Comment:
This has me a bit confused. Does the `db.UpdateEnrollment` API require that we "fetch" from the database (or that we already have the content from the database) that we will pass in as the `query` here? 

I mean, my intuition for an update API is that, given the necessary fields to uniquely fetch an existing row to update, the other fields would be updated, but not those that we didn't pass in.

Reply:
I think I have answered this question here: https://github.com/quickfeed/quickfeed/issues/570#issuecomment-1318468312. 
Using, for example, `Select("*").Updates(Updates(&qf.Enrollment{State: enrol.State, Status: enrol.Status, LastActivityDate: enrol.LastActivityDate}))` will update all the fields we don't "pass" to their default values. 

And because in different situations we only update one of these values, it is either `State`, `Status` or `LastActivityDate` that would be set in the query, and the two others would be "empty" before and if using `Select("*")` would be updated to their default values.